independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartCatalog.java

```diff
@@ -62,29 +61,12 @@ private QuarkusCodestartCatalog(Collection<Codestart> codestarts,
         this.extensionsMapping = extensionsMapping;
     }
 
-    public static QuarkusCodestartCatalog fromQuarkusPlatformDescriptor(QuarkusPlatformDescriptor platformDescriptor)
+    public static QuarkusCodestartCatalog fromExtensionsCatalog(ExtensionCatalog catalog, ResourceLoader resourceLoader)
             throws IOException {
-        final CodestartPathLoader pathLoader = platformPathLoader(platformDescriptor);
+        final CodestartPathLoader pathLoader = platformPathLoader(resourceLoader);
         final Collection<Codestart> codestarts = CodestartCatalogLoader.loadCodestarts(pathLoader, QUARKUS_CODESTARTS_DIR);
-        final Map<String, Extension> extensionsMapping = buildExtensionsMapping(platformDescriptor.getExtensions());
-        return new QuarkusCodestartCatalog(codestarts, extensionsMapping);
-    }
-
-    public static QuarkusCodestartCatalog fromQuarkusPlatformDescriptorAndDirectories(
```

Comment:
@aloubyansky this is used from outside..

Reply:
https://github.com/quarkusio/jbang-catalog/blob/master/src/QuarkusExampleCodestartDev.java#L140

Reply:
Well, QuarkusPlatformDescriptor does not exist any more. So the caller will have to call new methods.

Reply:
@aloubyansky yeah we just need to have the `ExtensionCatalog catalog, ResourceLoader resourceLoader, Collection<Path> directories` possibility

Reply:
It should be back now.